utils/post.js

```diff
@@ -216,13 +216,14 @@ const insertTopics = async (topics = []) => {
 
     const topic = topics[index];
     const topicIndex = parseInt(lastTopic.topic_id, 10) + parseInt(index, 10) + parseInt(1, 10);
+    const filterHashTagFromTopic = topic.replace(/#/g, '');
 
     try {
       await Topics.findOrCreate({
-        where: {name: topic},
+        where: {name: filterHashTagFromTopic},
         defaults: {
           topic_id: topicIndex,
-          name: topic,
+          name: filterHashTagFromTopic,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are correct in terms of removing '#' characters from the topic names before inserting them into the database. However, there's a potential issue with the generation of `topic_id`. If two requests come at the same time, they might end up generating the same `topic_id` which could lead to conflicts. Consider using a UUID or a sequence generated by the database itself for `topic_id` to avoid such issues. 

Also, you should handle the case where `Topics.findOrCreate` throws an error. Right now, if an error occurs, it will propagate up the call stack and might crash your application. You should catch the error and handle it appropriately.

```diff
- const topicIndex = parseInt(lastTopic.topic_id, 10) + parseInt(index, 10) + parseInt(1, 10);
+ const topicIndex = uuidv4(); // assuming you've imported uuidv4 from 'uuid'

try {
  await Topics.findOrCreate({
    where: {name: filterHashTagFromTopic},
    defaults: {
      topic_id: topicIndex,
      name: filterHashTagFromTopic,
      icon_path: '',
      is_custom_topic: true,
      created_at: new Date(),
    }
  });
+ } catch (error) {
+   console.error(`Failed to create topic: ${error}`);
+   // handle the error appropriately, e.g., by returning a response with an error status code
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->